src/components/widgets-settings/dates-picker/dates-picker.directive.coffee

```diff
@@ -109,6 +110,17 @@ module.directive('settingDatesPicker', ($templateCache, $filter, ImpacWidgetsSvc
             keep_today: isToToday()
         }
 
+      scope.onChange = ->
```

Comment:
The original `on-change` attribute called `scope.showApplyButton`, which encapsulates logic that is not applicable for the `scope.onChangeCb`. 

```
scope.showApplyButton = ->
        if scope.updateOnPick
          scope.applyChanges()
        else
          scope.changed = true
```

If I was to put the `scope.onChangeCb` inside the `scope.applyChanges` method, it would need to pass through the `scope.showApplyButton` method and which shouldn't be.

I think using `scope.onChange` to mediate the two scenarios makes sense.